Sources/NIO/ChannelPipeline.swift

```diff
@@ -173,6 +173,23 @@ public final class ChannelPipeline: ChannelInvoker {
         return promise.futureResult
     }
 
+    /// Adds the provided channel handlers to the pipeline in the order given, taking account
+    /// of the behaviour of `ChannelHandler.add(first:)`.
+    public func addHandlers(_ handlers: [ChannelHandler], first: Bool) -> EventLoopFuture<Void> {
+        var handlers = handlers
+        if first {
+            handlers = handlers.reversed()
+        }
+
+        return EventLoopFuture<Void>.andAll(handlers.map { add(handler: $0) }, eventLoop: eventLoop)
+    }
+
+    /// Adds the provided channel handlers to the pipeline in the order given, taking account
+    /// of the behaviour of `ChannelHandler.add(first:)`.
+    public func addHandlers(_ handlers: ChannelHandler..., first: Bool) -> EventLoopFuture<Void> {
```

Comment:
just a question but do we really need the variadic one?

Reply:
It would be a breaking change to remove it.

Reply:
(Although, unless NIOHTTP1 exports NIO, moving that method is also a breaking change)

Reply:
@tanner0101 but it was `private` before, right? So breaking it would be okay

Now obviously it's private as it's merged already but since we didn't tag yet we could still remove the variadic one. @normanmaurer / @Lukasa wdyt?

Reply:
Oh, duh. Yes you're right. I'm not at all opposed to removing it before the tag goes out.

Reply:
I don’t really care either way tbh. The variadic one seems like harmless syntactic sugar to me, but I’m certainly not motivated strongly by keeping it